src/gluonts/dataset/pandas.py

```diff
@@ -179,35 +207,22 @@ def from_long_dataframe(
         PandasDataset
             Gluonts dataset based on ``pandas.DataFrame``s.
         """
-        return cls(dataframes=dict(list(dataframe.groupby(item_id))), **kwargs)
+        return cls(dataframes=dataframe.groupby(item_id), **kwargs)
 
 
-def series_to_dataframe(
-    series: Union[pd.Series, List[pd.Series], Dict[str, pd.Series]]
-) -> Union[pd.DataFrame, List[pd.DataFrame], Dict[str, pd.DataFrame]]:
-    def to_df(series):
-        assert isinstance(
-            series.index, DatetimeIndexOpsMixin
-        ), "series index has to be a DatetimeIndex."
-        return series.to_frame(name="target")
+def pair_with_item_id(obj):
```

Comment:
Type annotation is missing.

I think this should use a `NamedTuple`. I think it's generally bad style to rely on some implied meaning through ordering.